File: app/src/main/java/com/jtrofe/cheesebots/physics/controllers/ParticleController.java
package com.jtrofe.cheesebots.physics.controllers;

import com.jtrofe.cheesebots.physics.Engine;
import com.jtrofe.cheesebots.physics.objects.GameObject;
import com.jtrofe.cheesebots.physics.objects.Particle;

/**
 * Track the life of particles and remove them when their life runs out
 */
public class ParticleController extends Controller{

    public ParticleController(Engine engine){
        super(engine);
    }

    @Override
    public void Update(double timeStep){

        for(GameObject obj:mEngine.Bodies){
            if(obj.Type != GameObject.TYPE_PARTICLE) continue;

            Particle p = (Particle) obj;

            p.LifeSpan --;

            if(p.LifeSpan <= 0){
                mEngine.RemoveBody(p);
            }
        }
    }
}

File: app/src/main/java/com/jtrofe/cheesebots/physics/controllers/ParticleController.kt
package com.jtrofe.cheesebots.physics.controllers

import com.jtrofe.cheesebots.physics.Engine
import com.jtrofe.cheesebots.physics.objects.GameObject
import com.jtrofe.cheesebots.physics.objects.Particle

/**
 * Created by MAIN on 2/11/16.
 */
public class ParticleController(engine:Engine):Controller(engine){

    override fun Update(timeStep:Double){

        mEngine.Bodies.filter{ it.Type == GameObject.TYPE_PARTICLE }
            .forEach{
                val p = it as Particle

                p.LifeSpan --

                if(p.LifeSpan <= 0){
                    mEngine.RemoveBody(p)
                }
            }
    }
}